### PR TITLE
Feat/create challenge #7

### LINKED ISSUE #7: add route to create a challenge

### PR DESCRIPTION
**Pull request by namadnuno**
Closes #7 

Added testing capability as well.

### How to test?
```
docker-compose exec apptest yarn test
```

### CODE REVIEW

File: routes/challenges.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const express = require('express')
+const router = express.Router()
+const { addChallenge } = require('../schema/ChallengeSchema')
```
Comment (rbgoncalves) [inline routes/challenges.js]: you should place all the requires first

File: index.js [hunk 2]
```diff
@@ -3,9 +3,10 @@ require('dotenv').config();
 const express = require('express')
 const app = express()
 const mongoose = require('mongoose')
-const sample = require('./routes/sample')
+const challenges = require('./routes/challenges')
```
Comment (rbgoncalves) [inline index.js]: you should place all the requires first

File: index.js [hunk 3]
```diff
@@ -3,9 +3,10 @@ require('dotenv').config();
...
+
+let DATABASE_URL = `mongodb://${process.env.MONGO_USERNAME}:${process.env.MONGO_PASSWORD}@${process.env.MONGO_HOST}:27017/${process.env.MONGO_DATABASE}`
```
Comment (rbgoncalves) [inline index.js]: you should use `const` in this case

File: test/challenges.test.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+var chai = require('chai')
+  , chaiHttp = require('chai-http');
+chai.use(chaiHttp);
+
+describe('challenge', () => {
+    beforeEach((cb) => db.collection('challenges').deleteMany({}, cb));
+    
+    it('should create a challange', (done) => {
+        let data = {
```
Comment (rbgoncalves) [inline test/challenges.test.js]: these data's should also be a `const` as you are not mutating them

File: test/challenges.test.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        })
+    })
+    it('should create a challange on request', (done) => {
+        let data = {
+            "description": "Do 100 push ups",
+            "hashtag": "#100_push_ups_against_covid",
+            "type": "media",
+            "url":
+            "https://media3.giphy.com/media/YMkht8VN9qi6eWjQC0/giphy.gif?cid=790b76115ba5b5f6bab4dbc48d0c9e8fceb0f4d2eb302239&rid=giphy.gif"
+        }
```
Comment (rbgoncalves) [inline test/challenges.test.js]: You could also use the same object instead of creating it twice

**Comment by namadnuno** [comment]
> Great!!
> 
> Just a few notes:
> 
>     * Please, remove yarn-error.log from Git ;-)
> 
>     * Could we use a `.edtorconfig` file?, since there are files with 2 or 4 indentation spaces I would add it with these configs:
> 
> 
> ```
> root = true
> 
> [*]
> indent_style = space
> indent_size = 2
> end_of_line = lf
> charset = utf-8
> trim_trailing_whitespace = true
> insert_final_newline = true
> ```
> 
> Great job, man ;-)

Thanks, @xppalmeida, ok, I have made that improvements.

**Comment by namadnuno** [comment]
> Good job! 👍 left some comments

I made that improvments, please check if everything is ok, and merge. 👊

**Pull request merged**